pkg/engine/vmextensions_test.go

```diff
@@ -367,3 +368,132 @@ func TestCreateAgentVMASCustomScriptExtension(t *testing.T) {
+			OrchestratorType: Kubernetes,
+		},
+		ExtensionProfiles: []*api.ExtensionProfile{
+			{
+				Name:    "winrm",
+				Version: "v1",
+				RootURL: "https://raw.githubusercontent.com/CecileRobertMichon/aks-engine/fix-extensions/",
+			},
+		},
+		AgentPoolProfiles: []*api.AgentPoolProfile{
+			{
+				Name:                "windowspool1",
+				OSType:              api.Windows,
+				AvailabilityProfile: "AvailabilitySet",
+				Extensions: []api.Extension{
+					{
+						Name: "winrm",
+					},
+				},
+			},
+		},
+	}
+
+	extensions := CreateCustomExtensions(properties)
+
+	expectedDeployments := []DeploymentARM{
+		{
+			DeploymentARMResource: DeploymentARMResource{
+				APIVersion: "[variables('apiVersionDeployments')]",
+				Copy: map[string]string{
+					"count": "[sub(variables('windowspool1Count'), variables('windowspool1Offset'))]",
+					"name":  "winrmExtensionLoop",
+				},
+				DependsOn: []string{"[concat('Microsoft.Compute/virtualMachines/', variables('windowspool1VMNamePrefix'), copyIndex(variables('windowspool1Offset')), '/extensions/cse', '-agent-', copyIndex(variables('windowspool1Offset')))]"},
+			},
+			DeploymentExtended: resources.DeploymentExtended{
+				Name: to.StringPtr("[concat(variables('windowspool1VMNamePrefix'), copyIndex(variables('windowspool1Offset')), 'winrm')]"),
+				Properties: &resources.DeploymentPropertiesExtended{
+					TemplateLink: &resources.TemplateLink{
+						URI:            to.StringPtr("https://raw.githubusercontent.com/CecileRobertMichon/aks-engine/fix-extensions/extensions/winrm/v1/template.json"),
+						ContentVersion: to.StringPtr("1.0.0.0"),
+					},
+					Parameters: map[string]interface{}{
+						"apiVersionDeployments": map[string]interface{}{"value": "[variables('apiVersionDeployments')]"},
+						"artifactsLocation":     map[string]interface{}{"value": "https://raw.githubusercontent.com/CecileRobertMichon/aks-engine/fix-extensions/"},
+						"extensionParameters":   map[string]interface{}{"value": "[parameters('winrmParameters')]"},
+						"targetVMName":          map[string]interface{}{"value": "[concat(variables('windowspool1VMNamePrefix'), copyIndex(variables('windowspool1Offset')))]"},
+						"targetVMType":          map[string]interface{}{"value": "agent"},
+						"vmIndex":               map[string]interface{}{"value": "[copyIndex(variables('windowspool1Offset'))]"},
+					},
+					Mode: resources.DeploymentMode("Incremental"),
+				},
+				Type: to.StringPtr("Microsoft.Resources/deployments"),
+			},
+		},
+	}
+
+	diff := cmp.Diff(extensions, expectedDeployments)
+
+	if diff != "" {
+		t.Errorf("unexpected diff while expecting equal structs: %s", diff)
+	}
+	properties = &api.Properties{
+		OrchestratorProfile: &api.OrchestratorProfile{
+			OrchestratorType: Kubernetes,
+		},
+		ExtensionProfiles: []*api.ExtensionProfile{
+			{
+				Name:    "hello-world-k8s",
+				Version: "v1",
+				RootURL: "https://raw.githubusercontent.com/CecileRobertMichon/aks-engine/fix-extensions/",
```

Comment:
I'll have to replace this in a follow up PR in order to make this unit test pass both in this PR and in master once it merges (the template gets downloaded from github and master has a different apiVersion variable).